pkg/dispatcher/dispatcher.go

```diff
@@ -89,14 +92,16 @@ func (d *Dispatcher) ListenOperationalState(operationalStateChannel <-chan event
 
 // RegisterDevice is a way for device synchronizers to register for
 // channel of config events
-func (d *Dispatcher) RegisterDevice(id topocache.ID) (chan events.ConfigEvent, error) {
+func (d *Dispatcher) RegisterDevice(id topocache.ID) (chan events.ConfigEvent, chan events.DeviceResponse, error) {
 	if d.deviceListeners[id] != nil {
-		return nil, fmt.Errorf("Device %s is already registered", id)
+		return nil, nil, fmt.Errorf("Device %s is already registered", id)
 	}
 	channel := make(chan events.ConfigEvent)
 	d.deviceListeners[id] = channel
-	log.Infof("Registering Device %s on channel w%v", id, channel)
-	return channel, nil
+	respChan := make(chan events.DeviceResponse)
```

Comment:
I thought you were going to create and delete this channel only when a request was made

Reply:
that was my assumption too, but then I would have no way of properly passing it down to the synchroniser.